src/store/DialogStore.js

```diff
@@ -1,11 +1,14 @@
 import { observable, action } from 'mobx';
 import { connect } from './Connect';
+import RoutingStore from './RoutingStore';
 
 const allDialogs = [];
 
 export default class DialogStore {
     constructor(mainStore) {
         this.mainStore = mainStore;
+        this.routingStore = new RoutingStore();
+        this.routingStore.registerDialog(this);
```

Comment:
@shayanbinary please use `this.mainStore.routing`, exactly the way you use it in `MenuStore.js`

Reply:
@aminroosta I try that also but return error `Uncaught RangeError: Maximum call stack size exceeded`